### PR TITLE
Fix handling of nested paths

### PR DESCRIPTION
**Pull request by mamcgrath**
Fixes handling of nested paths at the cost of adding a dependency on Lodash.

### CODE REVIEW

**Comment by jancama2** [comment]
@aditbiswas1 how does it look like with the MR? I kinda need this to be merged.

**Comment by ashbhir** [comment]
Approved

**Comment by jancama2** [comment]
@aditbiswas1 we need one more to merge it, could you pls find someone?

File: test/getStoreFromCookies.spec.js [hunk 1]
```diff
@@ -50,6 +54,7 @@ describe('getStateFromCookies', () => {
         // expectations
         expect(value.userId).toEqual(userId);
         expect(value.auth.authToken).toEqual(authToken);
+        expect(value.auth.key).toEqual(key);
```
Comment (ashbhir) [inline test/getStoreFromCookies.spec.js]: how is this test of `auth.key` any different from `auth.authToken` ?

Comment (jancama2) [reply test/getStoreFromCookies.spec.js]: With the current version of the library this would fail since there is no deep merge.

**Comment by ashbhir** [comment]
Approved

**Pull request merged**